activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -66,7 +66,7 @@ class StatementPool < ConnectionAdapters::StatementPool
       def initialize(connection, logger, connection_options, config)
         super(connection, logger, config)
 
-        @statements = StatementPool.new(self.class.type_cast_config_to_integer(config.fetch(:statement_limit) { 1000 }))
+        @statements = StatementPool.new(self.class.type_cast_config_to_integer(config.fetch(:statement_limit) { StatementPool::DEFAULT_STATEMENT_LIMIT }))
```

Comment:
We can remove the defaults from all these. `self.class.type_cast_config_to_integer(nil)` will return `nil`, so we can handle defaults in `StatementPool#initialize`.

```ruby
@statements = StatementPool.new(self.class.type_cast_config_to_integer(config[:statement_limit])
```